src/Client.ts

```diff
@@ -5,22 +5,32 @@ import {
 } from '@kontent-ai/delivery-sdk';
 import packageInfo from '../package.json';
 import { selectedProjectCookieName } from './const';
-import { defaultProjectId } from './Utilities/SelectedProject';
+import validator from 'validator';
 
 const sourceTrackingHeaderName = 'X-KC-SOURCE';
 
-// environment variables
-const projectId = process.env.REACT_APP_PROJECT_ID || '';
 const previewApiKey = process.env.REACT_APP_PREVIEW_API_KEY || '';
 
 const cookies = new Cookies(document.cookie);
 
-let currentProjectId = projectId || cookies.get(selectedProjectCookieName);
-if (currentProjectId) {
-  cookies.set(selectedProjectCookieName, currentProjectId, { path: '/' });
-} else {
-  currentProjectId = defaultProjectId;
-}
+const getProjectId = (): string => {

```

Comment:
In my opinion, it is nice to encode the possible "types of value" you return in the return type you declare. This function doesn't always return a project id. It sometimes fails to find one and returns an empty string, but this is not visible in the return type. Therefore I am not able to tell (from looking at the type definition when using the function somewhere) if it always finds a valid project id or not. If you'd return a `string | null` instead, it would be obvious that I have to handle a case where a project id just can't be found. It will require some additional handling in the call sites, but in the end, this is a good thing. It would make TypeScript watch you to always check for a missing id and, consequently, make the code more resilient to bugs.